### PR TITLE
fix swallowing for groups

### PR DESCRIPTION
**Pull request by Aqa-Ib**
#### Describe your PR, what does it fix/add?
Fixes https://github.com/hyprwm/Hyprland/issues/8206

#### Is it ready for merging, or does it need work?
Ready.

### CODE REVIEW

**Comment by Aqa-Ib** [comment]
I'm gonna refactor this PR.

**Comment by Aqa-Ib** [comment]
In the end, this is the best approach that I could achieve.
Ready to merge if you like it.

**Comment by vaxerski** [comment]
I'll read this tomorrow. Generally, windows should be separated from containers internally in the codebase, heh, but it's way too late for that.

**Comment by Aqa-Ib** [comment]
> Generally, windows should be separated from containers internally in the codebase, heh, but it's way too late for that.

Better late than never, we can move the swallowing logic to onWindowCreated and onWindowDestroyed if that's what you mean ?

**Comment by vaxerski** [comment]
yeah at this point it'd be really hard + break a lot of plugins, plus prolly introduce tons of bugs. Shit's... good enough I guess. I don't think we'll be doing anything fancier than groups on windows.

File: src/layout/IHyprLayout.cpp [hunk 1]
```diff
@@ -193,7 +193,11 @@ bool IHyprLayout::onWindowCreatedAutoGroup(PHLWINDOW pWindow) {
     if (pWindow->m_bIsFloating && !OPENINGON->m_bIsFloating)
         denied = true;
 
-    if (*PAUTOGROUP                                      // check if auto_group is enabled.
+    bool swallowing = false;
+    if (pWindow->m_pSwallowed || pWindow->m_bGroupSwallowed)
+        swallowing = true;
```
Comment (vaxerski) [inline src/layout/IHyprLayout.cpp:198]: `const bool SWALLOWING = pWindow->m_pSwallowed || pWindow->m_bGroupSwallowed;` bud..?

Comment (Aqa-Ib) [reply src/layout/IHyprLayout.cpp]: Thanks. I have also changed the other variables too for consistency.

**Review by vaxerski — APPROVED** [review]

**Pull request merged**